src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -1873,10 +1898,149 @@ public void testEventStreamPolling() throws Exception {
     assertNotNull(eventList);
     assertThat(eventList, not(empty()));
 
-    final Event pullEvent = eventList.get(0);
-    final Event createEvent = eventList.get(1);
-    final Event startEvent = eventList.get(2);
-    eventStreamAssertions(pullEvent, createEvent, startEvent, container.id(), BUSYBOX_LATEST);
+    imagePullEventAssertions(eventList.get(0));
+
+    // create and start event assertions
+    containerEventAssertions(eventList.get(1), containerId, containerName,
+            "create", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(2), containerId, containerName,
+            "start", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(3), containerId, containerName,
+            "die", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(4), containerId, containerName,
+            "destroy", BUSYBOX_LATEST);
+  }
+
+  @Test(timeout = 10000)
+  public void testEventTypes() throws Exception {
+    requireDockerApiVersionAtLeast("1.22", "Event types");
+
+    final String volumeName = randomName();
+    final String containerName = randomName();
+    final String mountPath = "/anywhere";
+    final Volume volume = Volume.builder().name(volumeName).build();
+    final HostConfig hostConfig = HostConfig.builder()
+            .binds(Bind.from(volume).to(mountPath).build())
+            .build();
+    final ContainerConfig config = ContainerConfig.builder()
+            .image(BUSYBOX_LATEST)
+            .hostConfig(hostConfig)
+            .build();
+
+    // Wait once to clean our event "palette" of events from other tests
+    Thread.sleep(1000);
+    final Date start = new Date();
+    final long startTime = start.getTime() / 1000;
+
+    sut.pull(BUSYBOX_LATEST);
+    sut.createVolume(volume);
+    final ContainerCreation container = sut.createContainer(config, containerName);
+    final String containerId = container.id();
+    sut.startContainer(containerId);
+    Thread.sleep(1000);
```

Comment:
That is correct. Writing these tests, I got into problems when I tried to remove a container that was still running. I also tried to stop the container first, but sometimes it was stopped already so that caused its own problems.